src/book/mod.rs

```diff
@@ -133,69 +137,44 @@ impl MDBook {
 
         debug!("[fn]: init");
 
-        if !self.config.get_root().exists() {
-            fs::create_dir_all(&self.config.get_root()).unwrap();
-            info!("{:?} created", &self.config.get_root());
-        }
-
         {
-
-            if !self.get_destination().exists() {
-                debug!("[*]: {:?} does not exist, trying to create directory", self.get_destination());
-                fs::create_dir_all(self.get_destination())?;
-            }
-
-
-            if !self.config.get_source().exists() {
-                debug!("[*]: {:?} does not exist, trying to create directory", self.config.get_source());
-                fs::create_dir_all(self.config.get_source())?;
+            let root = self.config.get_root();
+            let dest = self.get_destination();
+            let src = self.config.get_source();
+
+            let necessary_folders = &[root, dest, src];
+            
+            for folder in necessary_folders {
+                if !folder.exists() {
+                    fs::create_dir_all(folder)?;
+                    debug!("{} created", folder.display());
+                }
             }
 
-            let summary = self.config.get_source().join("SUMMARY.md");
+            let summary = src.join("SUMMARY.md");
 
             if !summary.exists() {
+                debug!("[*]: Creating SUMMARY.md");
 
-                // Summary does not exist, create it
-                debug!("[*]: {:?} does not exist, trying to create SUMMARY.md", &summary);
                 let mut f = File::create(&summary)?;
 
-                debug!("[*]: Writing to SUMMARY.md");
-
                 writeln!(f, "# Summary")?;
-                writeln!(f, "")?;
+                writeln!(f)?;
```

Comment:
how about writing it all in one go with one call?

Also I would suggest having these strings stored somewhere in one place as constants instead hardcoding the values along the code This will make it easier to change and inspect later on.

Reply:
I like this idea. I've pulled them out into a `const` string at the very top so we can change them easily later on.